enhancements/update/cluster-version-operator-x509-upstream-trust.md

```diff
@@ -0,0 +1,202 @@
+---
+title: cluster-version-operator-x.509-upstream-trust
+authors:
+  - "@wking"
+reviewers:
+  - "@LalatenduMohanty"
+approvers:
+  - "@bparees"
+  - "@deads2k"
+creation-date: 2020-05-12
+last-updated: 2020-05-14
+status: provisional
+see-also:
+  - "/enhancements/automated-service-ca-rotation.md"
+  - "/enhancements/kube-apiserver/certificates.md"
+  - "/enhancements/kube-apiserver/tls-config.md"
+  - "/enhancements/proxy/global-cluster-egress-proxy.md"
+  - "https://github.com/openshift/enhancements/pull/115"
+---
+
+# Cluster-version Operator X.509 Upstream Trust
+
+This enhancement provides a mechanism for providing [the cluster-version operator][cluster-version-operator] (CVO) with an alternative trust-store and TLS configuration to use when connecting to [the update recommendation service][update-service] that overrides both the CVO's defaults and any [proxy configuration](../proxy/global-cluster-egress-proxy.md).
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [x] Test plan is defined
+- [x] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Summary
+
+The CVO retrieves available updates from [an update recommendation service][update-service] serving [the Cincinnati protocol][cincinnati].
+The CVO also retrieves release image signatures from [configured stores][update-keys] (and [other places](../oc/mirroring-release-image-signatures.md)) to establish trust for the recommended update targets.
+Both of those may require HTTPS calls to external services.
+
+In some cases, those HTTPS calls will use a [configured proxy](../proxy/global-cluster-egress-proxy.md).
+And currently there is no mechanism for configuring additional HTTP(S) signature stores.
+But the update recommendation service is configurable [with ClusterVersion's `upstream` property][api-upstream], and adminstrators may point that at internal services whose X.509 certificate is not signed by a certificate authority (CA) that the CVO trusts by default.
+
+## Motivation
+
```

Comment:
Communication to any https endpoint requires three logical things

1. destination - this is usually a URL
2. means to verify the destination - this is a CA bundle
3. means to identify client/self to the destination - this is either a bearer token and/or a client cert/key pair

This is true of communication to any endpoint by any client.  Not all endpoints imply the same level trust and requiring admins to share CA bundles means that the security relies on the least protected key.  One concrete example: a service serving CA bundle should not be trusted for names that are not services.

Reply:
> Communication to any https endpoint requires three logical things...

Added the TLS security profile as a fourth thing, and twiddled with the language slightly, with c5ba370 -> b6105c7.